plugins/kinetic-dp/fu-kinetic-dp-plugin.c

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (C) 2017 Mario Limonciello <mario.limonciello@dell.com>
+ * Copyright (C) 2017 Peichen Huang <peichenhuang@tw.kinetic.com>
+ * Copyright (C) 2017 Richard Hughes <richard@hughsie.com>
+ * Copyright (C) 2022 Hai Su <hsu@kinet-ic.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-kinetic-dp-plugin.h"
+#include "fu-kinetic-dp-puma-device.h"
+#include "fu-kinetic-dp-puma-firmware.h"
+#include "fu-kinetic-dp-secure-device.h"
+#include "fu-kinetic-dp-secure-firmware.h"
+#include "fu-kinetic-dp-struct.h"
+
+struct _FuKineticDpPlugin {
+	FuPlugin parent_instance;
+};
+
+G_DEFINE_TYPE(FuKineticDpPlugin, fu_kinetic_dp_plugin, FU_TYPE_PLUGIN)
+
+static FuKineticDpDevice *
+fu_kinetic_dp_plugin_create_device(FuDpauxDevice *dpaux_device, GError **error)
+{
+	FuKineticDpChip chip_id = 0;
+	FuKineticDpFwState fw_state = 0;
+	const gchar *dev_id = fu_dpaux_device_get_dpcd_dev_id(dpaux_device);
+	g_autoptr(FuKineticDpDevice) dp_device = NULL;
+	g_autofree gchar *str = NULL;
+	const struct {
+		FuKineticDpChip chip_id;
+		FuKineticDpFwState fw_state;
+		const gchar *id_str;
+	} map[] = {{
+		       FU_KINETIC_DP_CHIP_JAGUAR_5000,
+		       FU_KINETIC_DP_FW_STATE_IROM,
+		       "5010IR",
+		   },
+		   {
+		       FU_KINETIC_DP_CHIP_JAGUAR_5000,
+		       FU_KINETIC_DP_FW_STATE_APP,
+		       "KT50X0",
+		   },
+		   {
+		       FU_KINETIC_DP_CHIP_MUSTANG_5200,
+		       FU_KINETIC_DP_FW_STATE_IROM,
+		       "5210IR",
+		   },
+		   {
+		       FU_KINETIC_DP_CHIP_MUSTANG_5200,
+		       FU_KINETIC_DP_FW_STATE_APP,
+		       "KT52X0",
+		   },
+		   {
+		       FU_KINETIC_DP_CHIP_MUSTANG_5200,
+		       FU_KINETIC_DP_FW_STATE_APP,
+		       "KT5200",
+		   },
+		   {FU_KINETIC_DP_CHIP_PUMA_2900, FU_KINETIC_DP_FW_STATE_IROM, "PUMA"},
```

Comment:
did clang do this funny whitespace compared to the rest?

Reply:
Yes! I'll see if I can coerce it to something sane